floris/floris_model.py

```diff
@@ -973,98 +1064,36 @@ def get_farm_AEP(
                 objective function. If None, this  is an array with all values
                 1.0 and with shape equal to (n_findex,
                 n_turbines). Defaults to None.
-            no_wake: (bool, optional): When *True* updates the turbine
-                quantities without calculating the wake or adding the wake to
-                the flow field. This can be useful when quantifying the loss
-                in AEP due to wakes. Defaults to *False*.
-
-
-        Returns:
-            float:
-                The Annual Energy Production (AEP) for the wind farm in
-                watt-hours.
         """
 
-        # Verify dimensions of the variable "freq"
-        if np.shape(freq)[0] != self.core.flow_field.n_findex:
-            raise UserWarning(
-                "'freq' should be a one-dimensional array with dimensions (n_findex). "
-                f"Given shape is {np.shape(freq)}"
-            )
-
-        # Check if frequency vector sums to 1.0. If not, raise a warning
-        if np.abs(np.sum(freq) - 1.0) > 0.001:
-            self.logger.warning(
-                "WARNING: The frequency array provided to get_farm_AEP() does not sum to 1.0."
-            )
+        farm_power = self._get_farm_power(turbine_weights=turbine_weights)
 
-        # Copy the full wind speed array from the floris object and initialize
-        # the the farm_power variable as an empty array.
-        wind_speeds = np.array(self.core.flow_field.wind_speeds, copy=True)
-        wind_directions = np.array(self.core.flow_field.wind_directions, copy=True)
-        turbulence_intensities = np.array(self.core.flow_field.turbulence_intensities, copy=True)
-        farm_power = np.zeros(self.core.flow_field.n_findex)
-
-        # Determine which wind speeds we must evaluate
-        conditions_to_evaluate = wind_speeds >= cut_in_wind_speed
-        if cut_out_wind_speed is not None:
-            conditions_to_evaluate = conditions_to_evaluate & (wind_speeds < cut_out_wind_speed)
-
-        # Evaluate the conditions in floris
-        if np.any(conditions_to_evaluate):
-            wind_speeds_subset = wind_speeds[conditions_to_evaluate]
-            wind_directions_subset = wind_directions[conditions_to_evaluate]
-            turbulence_intensities_subset = turbulence_intensities[conditions_to_evaluate]
-            self.set(
-                wind_speeds=wind_speeds_subset,
-                wind_directions=wind_directions_subset,
-                turbulence_intensities=turbulence_intensities_subset,
-            )
-            if no_wake:
-                self.run_no_wake()
+        if freq is None:
+            if self.wind_data is None:
+                freq = np.array([1.0])
```

Comment:
to return the mean farm power, should this be freq = np.array([1.0/self.core.flow_field.n_findex])?